### PR TITLE
Add json and py to editorconfig

### PR DESCRIPTION
**Pull request by warp-core**
**Development environment**

## Summary
Adds an entry to the .editorconfig file for files matching the pattern `*.json` that tells it to use a tab for each level of indentation, so VS Code (and any other editorconfig enabled text editor) uses the correct indentation style.

Also, changes some cmake and vcpkg related json files to use tabs instead of pairs of spaces for indentation.

Also add an entry for ".py" files. Tabs are used for indentation.
We currentlyhave four Python files in the repository. Two use tabs for indentation, two use spaces.
This PR modifies the two that use spaces to use tabs.

Also, removed the editorconfig entry for the catch.hpp file since we no longer have that.

## Testing Done
Open a json file within the repository in VS Code and it no longer highlights some of the indentations.
It adds and removes tabs for indentation correctly and highlights space based indentations as incorrect.

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -49,3 +49,7 @@ indent_size = 2
 [*.{yml,yaml}]
 indent_size = 2
 indent_style = space
+
+[*.json]
+indent_size = 2
+indent_style = space
```
Comment (warp-core) [inline .editorconfig:55]:
```suggestion
indent_style = space

[utils.contentStyle.json]
indent_style = tabs
```

File: .editorconfig [hunk 2]
```diff
@@ -49,3 +49,10 @@ indent_size = 2
...
+
+[utils.contentStyle.json]
```
Comment (warp-core) [inline .editorconfig:57]:
```suggestion
[utils/contentStyle.json]
```

**Comment by mOctave** [comment]
Should we just change the indentation for `contentStyle.json` so it matches?

**Comment by warp-core** [comment]
> Should we just change the indentation for `contentStyle.json` so it matches?

I didn't want to lead with changing every line in that file, but that is an option, yes.

**Comment by RisingLeaf** [comment]
> > Should we just change the indentation for `contentStyle.json` so it matches?
> 
> I didn't want to lead with changing every line in that file, but that is an option, yes.

I think our files should match our style, and you probably can just apply the defined style, right?
So I would say do it

**Review by tibetiroka — COMMENTED** [review]
Can you also remove the catch.hpp entry from above?

**Review by petervdmeer — APPROVED** [review]

**Comment by warp-core** [comment]
Alternatively, should the CMake configurations (and some other things) be changed to use tabs? All the same reasons for using tabs in the C++ source code and game data are just as applicable to these things.

File: .editorconfig [hunk 3]
```diff
@@ -49,3 +46,10 @@ indent_size = 2
 [*.{yml,yaml}]
 indent_size = 2
 indent_style = space
+
+[*.json]
+indent_size = 2
+indent_style = space
+
+[utils/contentStyle.json]
+indent_style = tabs
```
Comment (TomGoodIdea) [inline .editorconfig:55]:
```suggestion
indent_style = tab
```

Comment (warp-core) [reply .editorconfig]: This line no longer exists.

File: utils/contentStyle.json [hunk 4]
```diff
@@ -33,7 +33,7 @@
 				"checks": [
 					{
 						"description": "multiple consecutive space characters",
-						"regex": "(?<!^tip .*)([^ ])  +([^# ])",
+						"regex": "(?<!^tip .*)([^ ])	+([^# ])",
```
Comment (warp-core) [inline utils/contentStyle.json:36]:
```suggestion
						"regex": "(?<!^tip .*)([^ ])    +([^# ])",
```

File: utils/contentStyle.json [hunk 5]
```diff
@@ -33,7 +33,7 @@
 				"checks": [
 					{
 						"description": "multiple consecutive space characters",
-						"regex": "(?<!^tip .*)([^ ])  +([^# ])",
+						"regex": "(?<!^tip .*)([^ ])    +([^# ])",
```
Comment (warp-core) [inline utils/contentStyle.json:36]:
```suggestion
						"regex": "(?<!^tip .*)([^ ])  +([^# ])",
```

**Comment by warp-core** [comment]
This PR now also adds an entry to the editorconfig for .py files and changes two .py files from using spaces to using tabs for indentation.

File: utils/korath-cipher.py [hunk 6]
```diff
@@ -35,7 +35,7 @@
 #
 # ORIGINAL: Dalam kebenaran, saya hanya melihat kebohongan.
 # REVERSED: malaD ,naranebek ayas aynah tahilem .nagnohobek
-# EXILE:    famaS ,ralarehet aga' agrap kapimef .ranrupuhet
+# EXILE:	famaS ,ralarehet aga' agrap kapimef .ranrupuhet
```
Comment (mOctave) [inline utils/korath-cipher.py:38]:
```suggestion
# EXILE: famaS ,ralarehet aga' agrap kapimef .ranrupuhet
```

File: utils/korath-cipher.py [hunk 7]
```diff
@@ -58,12 +58,12 @@
...
-# Quarg => Kuwaru                               (Efreti)
-# Drak => Drak                                  (both Exile and Efreti)
-# Ember Space => Nraol Alaj                     (Exile)
-# There Might Be Riots => Ter Mite Bee Riot     (both Exile and Efreti)
+# humans => humanika							(Exile, Efreti, and Quarg)
+# human (noun, member of species) => humani	 (Exile, Efreti, and Quarg)
+# Quarg => Kuwaru							   (Efreti)
+# Drak => Drak								  (both Exile and Efreti)
+# Ember Space => Nraol Alaj					 (Exile)
+# There Might Be Riots => Ter Mite Bee Riot	 (both Exile and Efreti)
```
Comment (mOctave) [inline utils/korath-cipher.py:66]:
```suggestion
# humans => humanika                          (Exile, Efreti, and Quarg)
# human (noun, member of species) => humani   (Exile, Efreti, and Quarg)
# Quarg => Kuwaru                             (Efreti)
# Drak => Drak                                (both Exile and Efreti)
# Ember Space => Nraol Alaj                   (Exile)
# There Might Be Riots => Ter Mite Bee Riot   (both Exile and Efreti)
```

File: utils/korath-cipher.py [hunk 8]
```diff
@@ -85,58 +85,58 @@
...
-    # Print the results of this line:
-    print("ORIGINAL: "+" ".join(words))
-    print("REVERSED: "+" ".join(sdrow))
-    print("EXILE:    "+" ".join(exiles))
-    print("EFRETI:   "+" ".join(efretis))
-    print("")
+	# Print the results of this line:
+	print("ORIGINAL: "+" ".join(words))
+	print("REVERSED: "+" ".join(sdrow))
+	print("EXILE:	"+" ".join(exiles))
```
Comment (mOctave) [inline utils/korath-cipher.py:140]:
```suggestion
	print("EXILE:    "+" ".join(exiles))
```

File: utils/korath-cipher.py [hunk 9]
```diff
@@ -85,58 +85,58 @@
...
 # Loop over every line in stdin, processing it
 for line in sys.stdin:
-    words = line.split()       # words in the line, as a list
-    sdrow = []                 # the same words, with letters reversed
-    exiles = []                # those words after the exile cipher
-    efretis = []               # those words after the efret cipher
+	words = line.split()	   # words in the line, as a list
+	sdrow = []				 # the same words, with letters reversed
+	exiles = []				# those words after the exile cipher
+	efretis = []			   # those words after the efret cipher
```
Comment (mOctave) [inline utils/korath-cipher.py:112]:
```suggestion
	words = line.split() # words in the line, as a list
	sdrow = [] # the same words, with letters reversed
	exiles = [] # those words after the exile cipher
	efretis = [] # those words after the efret cipher
```

**Review by mOctave — COMMENTED** [review]
There's a few places here where a bulk space->tab conversion broke things; I think my suggestions would work better.

File: utils/korath-cipher.py (hunk 9 above)
Comment (warp-core) [reply utils/korath-cipher.py:112]:
```suggestion
	words = line.split()       # words in the line, as a list
	sdrow = []                 # the same words, with letters reversed
	exiles = []                # those words after the exile cipher
	efretis = []               # those words after the efret cipher
```

File: utils/korath-cipher.py (hunk 6 above)
Comment (warp-core) [reply utils/korath-cipher.py:38]:
```suggestion
# EXILE:    famaS ,ralarehet aga' agrap kapimef .ranrupuhet
```

**Review by tibetiroka — APPROVED** [review]

**Comment by xobes** [comment]
@warp-core 
re:
> Also add an entry for ".py" files. Tabs are used for indentation.
> We currentlyhave four Python files in the repository. Two use tabs for indentation, two use spaces.
> This PR modifies the two that use spaces to use tabs

Why wouldn't use use spaces to align with PEP-8:  https://peps.python.org/pep-0008/#tabs-or-spaces

**Comment by petervdmeer** [comment]
We only have 4 Python scripts at this moment, and this project doesn't have specific coding style guidelines to align with PEP-8 or not. I'm okay with merging this as-is now and switch over to PEP-8 at a later moment (if we decide that want to do that in the future).

**Pull request merged**